### PR TITLE
chore: enable conditional amplify app creation during init for testing purposes

### PR DESCRIPTION
**Pull request by attilah**
*Description of changes:*

Add logic to the  CLI to enable e2e tests to control whether to create an Amplify Console App or not.

E2E tests can specify the `disableAmplifyAppCreation` option parameter for the `initJSProjectWithProfile` method for `amplify init`, so CLI will not create the Amplify Console App automatically. It enables the CLI team to test special use cases when presence of Amplify Console App is required or not. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts [hunk 1]
```diff
@@ -72,7 +81,13 @@ export function initJSProjectWithProfile(cwd: string, settings: Object) {
...
   const s = { ...defaultSettings, ...settings };
   return new Promise((resolve, reject) => {
-    spawn(getCLIPath(), ['init'], { cwd, stripColors: true })
+    spawn(getCLIPath(), ['init'], {
+      cwd,
+      stripColors: true,
+      env: {
+        CLI_DEV_INTERNAL_DISABLE_AMPLIFY_APP_CREATION: '1',
+      },
+    })
```
Comment (yuth) [inline packages/amplify-e2e-core/src/init/initProjectHelper.ts:90]: Why is this hardcoded?

**Review by yuth — APPROVED** [review]

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts [hunk 2]
```diff
@@ -33,8 +34,16 @@ export const amplifyRegions = [
 
 export function initJSProjectWithProfile(cwd: string, settings: Object) {
   const s = { ...defaultSettings, ...settings };
+  let env;
+
+  if (s.disableAmplifyAppCreation === true) {
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-core/src/init/initProjectHelper.ts:39]: Are these circleCI settings?

File: packages/amplify-e2e-tests/src/__tests__/api_2.test.ts [hunk 3]
```diff
@@ -21,7 +19,7 @@ import {
   getProjectMeta,
   getTransformConfig,
 } from 'amplify-e2e-core';
-import { TRANSFORM_CURRENT_VERSION, TRANSFORM_BASE_VERSION, writeTransformerConfiguration } from 'graphql-transformer-core';
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-tests/src/__tests__/api_2.test.ts:24]: Is this change a part of this PR?

File: packages/amplify-e2e-tests/typings/aws-matchers.d.ts [hunk 4]
```diff
@@ -2,5 +2,6 @@ namespace jest {
   interface Matchers<R> {
     toBeIAMRoleWithArn(roleName: string, arn?: string): R;
     toBeAS3Bucket(bucketName: string): R;
+    toHaveValidPolicyConditionMatchingIdpId(idpId: string): R;
```
Comment (kaustavghosh06) [inline packages/amplify-e2e-tests/typings/aws-matchers.d.ts:5]: What's this change tied to?

File: packages/amplify-provider-awscloudformation/src/amplify-service-manager.js [hunk 5]
```diff
@@ -415,6 +419,8 @@ async function uploadFile(s3, filePath, key) {
   }
 }
 
+const amplifyAppCreationEnabled = () => !process.env || process.env.CLI_DEV_INTERNAL_DISABLE_AMPLIFY_APP_CREATION !== '1';
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/src/amplify-service-manager.js:422]: Why not boolean? Can you document the behavior of this env variable in the description please?

**Review by kaustavghosh06 — COMMENTED** [review]
This PR has a couple of improvements and refactors which are great but not relevant to the conditional amplify-app testing use-case. Maybe consider separating the refactor from the conditional amplify app updates PR?

File: packages/amplify-e2e-core/src/init/initProjectHelper.ts (hunk 2 above)
Comment (attilah) [reply packages/amplify-e2e-core/src/init/initProjectHelper.ts:39]: No, this is the option tests use to control wether the env var will be set for launched CLI or not.

File: packages/amplify-e2e-tests/src/__tests__/api_2.test.ts (hunk 3 above)
Comment (attilah) [reply packages/amplify-e2e-tests/src/__tests__/api_2.test.ts:24]: Yes they were unused imports and tsc complains about them if you build the tests manually with yarn tsc and the e2e test package had build failures, I fixed those to make sure code is correct.

File: packages/amplify-e2e-tests/typings/aws-matchers.d.ts (hunk 4 above)
Comment (attilah) [reply packages/amplify-e2e-tests/typings/aws-matchers.d.ts:5]: When this matcher was added it was not added to .d.ts and TSC requires it to be present, e2e test package had build failures, I fixed those to make sure code is correct.

File: packages/amplify-provider-awscloudformation/src/amplify-service-manager.js (hunk 5 above)
Comment (attilah) [reply packages/amplify-provider-awscloudformation/src/amplify-service-manager.js:422]: 0 and 1 is a common practice for env vars to represent booleans, in most of the cases code just checks if env var is present or not. I wanted to have a strict check on the value to reduce the possibility of misconfiguration.
I add more documentation to the PR description.

**Review by kaustavghosh06 — APPROVED** [review]

**Pull request merged**